### PR TITLE
[BS-1] Add Data layer entities

### PR DESCRIPTION
**Pull request by VladimirBrejcha**
**Description**

Added base data layer entities:
- Category
- Tag
- Account
- Transaction

**Affects**

Data layer

### CODE REVIEW

File: bonsai.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -74,6 +82,7 @@
 				20EC47A5275D14DC00881E0F /* SwiftGen */,
 				20115259275BCF5E003BEF0B /* Appearance */,
 				20A602FB275BB17000745114 /* Home */,
+				7E3262D1275E23D10045CEA0 /* Data */,
```
Comment (antonhoang) [inline bonsai.xcodeproj/project.pbxproj]: I think it might be better rename Data folder to Entities

File: bonsai/Data/Entity/Transaction.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//  bonsai
+//
+//  Created by Vladimir Korolev on 06.12.2021.
+//
+
+import Foundation
+
+struct Transaction: Identifiable {
+    let id: UUID
+
```
Comment (antonhoang) [inline bonsai/Data/Entity/Transaction.swift:88]: no need empty space

File: bonsai.xcodeproj/project.pbxproj (hunk 1 above)
Comment (VladimirBrejcha) [reply bonsai.xcodeproj/project.pbxproj]: This one is in progress. I think we should discuss it together later in context of app architecture

File: bonsai/Data/Entity/Transaction.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    let account: Account
+    let amount: Double
+    let title: String?
+    let category: Category
+    let tags: [Tag]
+    let date: Date
+}
+
+extension Transaction {
+    enum Kind {
```
Comment (antonhoang) [inline bonsai/Data/Entity/Transaction.swift]: 'Type' have more sense than 'Kind' of transaction

**Comment by VladimirBrejcha** [comment]
test comment

**Comment by VladimirBrejcha** [comment]
test comment 2

File: bonsai/Data/Entity/Category.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//  bonsai
+//
+//  Created by Vladimir Korolev on 06.12.2021.
+//
+
+import Foundation
+
+struct Category: Identifiable {
+    let id: UUID
+    let title: String
```
Comment (Valter4578) [inline bonsai/Data/Entity/Category.swift]: also we need to add icon and color here

File: bonsai/Data/Entity/Transaction.swift (hunk 2 above)
Comment (VladimirBrejcha) [reply bonsai/Data/Entity/Transaction.swift:88]: Empty space here separates Identifiable conformance from actual entity

I think we should not discuss code-style in the PR's but instead focus on logic, bugs etc

Comment (Valter4578) [reply bonsai/Data/Entity/Transaction.swift:88]: Agree here. I suppose empty spaces aren't really important considering that we're building MVP

**Comment by VladimirBrejcha** [comment]
test comment

File: bonsai/Data/Entity/Account.swift [hunk 5]
```diff
@@ -0,0 +1,13 @@
+//
+//  Account.swift
+//  bonsai
+//
+//  Created by Vladimir Korolev on 06.12.2021.
+//
+
+import Foundation
+
+struct Account: Identifiable {
```
Comment (VladimirBrejcha) [inline bonsai/Data/Entity/Account.swift:10]: Test

File: bonsai/Data/Entity/Category.swift (hunk 4 above)
Comment (VladimirBrejcha) [reply bonsai/Data/Entity/Category.swift]: added

File: bonsai/Data/Entity/Transaction.swift (hunk 3 above)
Comment (VladimirBrejcha) [reply bonsai/Data/Entity/Transaction.swift]: fixed

File: bonsai.xcodeproj/project.pbxproj (hunk 1 above)
Comment (VladimirBrejcha) [reply bonsai.xcodeproj/project.pbxproj]: fixed

File: bonsai/Entity/Category.swift [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by Vladimir Korolev on 06.12.2021.
+//
+
+import UIKit
+
+struct Category: Identifiable {
+    let id: UUID
+    let title: String
+    let color: UIColor
```
Comment (Valter4578) [inline bonsai/Entity/Category.swift]: We don't write UIKit app so u need to change UIKit types to SwiftUI's ones

**Review by antonhoang — APPROVED** [review]

**Review by Valter4578 — APPROVED** [review]

**Pull request merged**